src/vivarium/framework/results/observer.py

```diff
@@ -28,12 +34,19 @@ def setup_component(self, builder: Builder) -> None:
 
     def get_report_attributes(self, builder: Builder) -> None:
         """Define commonly-used attributes for reporting."""
-        try:
-            self.results_dir = builder.configuration.output_data.results_directory
-        except KeyError:
-            self.results_dir = None
-        self.input_draw = builder.configuration.input_data.input_draw_number
-        self.random_seed = builder.configuration.randomness.random_seed
+        self.results_dir = (
+            builder.configuration.to_dict()
```

Comment:
Nit: you can save `config_dict = builder.configuration.to_dict()` so you only have to make that call once.